source-azure-blob-storage/main.go

```diff
@@ -228,14 +241,20 @@ func (l *azureBlobListing) Next() (filesource.ObjectInfo, error) {
 
 		obj := filesource.ObjectInfo{}
 
-		obj.Path = *blob.Name
+		obj.Path = filesource.PartsToPath(l.container, *blob.Name)
 		obj.ModTime = *blob.Properties.LastModified
 		obj.ContentType = *blob.Properties.ContentType
 		obj.Size = *blob.Properties.ContentLength
 
 		if blob.Properties.ContentEncoding != nil {
 			obj.ContentEncoding = *blob.Properties.ContentEncoding
 		}
+
+		if l.query.StartAt != "" && obj.Path < l.query.StartAt {
+			log.WithField("path", obj.Path).Debug("skipping blob before StartAt")
```

Comment:
How many logs is too many?  Should we count the number of skipped paths and log only the count instead of each file?